### PR TITLE
fix(suite-common): remove sort from getAccountTransactions

### PR DESCRIPTION
**Pull request by komret**
Display correct transactions when directly navigating to a page in transaction history.

## Description

- Removes sorting from `getAccountTransactions`. The list contains null values as placeholders for transactions that have not been fetched yet. Sorting the whole list pushes null values to the end which causes transactions to not appear on the correct page until all previous transactions are fetched.

- Adds sorting to `analyzeTransactions` where it is required as explained in #5785 (the bug was also introduced here)

## Screenshots (if appropriate):
Screenshot 2022-08-19 at 13 15 38

@szymonlesisz, can you please check this does not break your changes from 751d3a2c22fb998dd5b9642a7d410d2584da5798?

**QA**: Please check pagination works correctly and the changes should not cause the error described in #2643. To reproduce the bug, there must be 51+ transactions in the account and you need to navigate to page 3 or more before visiting other pages.

### CODE REVIEW

File: suite-common/wallet-utils/src/transactionUtils.ts [hunk 1]
```diff
@@ -23,14 +23,15 @@ export const sortByBlockHeight = (a: WalletAccountTransaction, b: WalletAccountT
...
+ */
 export const getAccountTransactions = (
     accountKey: string,
     transactions: Record<string, WalletAccountTransaction[]>,
-) => {
-    const txs = transactions[accountKey] || [];
-    // clone array (read-only from reducer) and sort by blockHeight
-    return txs.slice(0).sort(sortByBlockHeight);
-};
+) => transactions[accountKey] || [];
```
Comment (dahaca) [inline suite-common/wallet-utils/src/transactionUtils.ts:34]: Perhaps you could add a null check for `transactions` here?

Comment (dahaca) [inline suite-common/wallet-utils/src/transactionUtils.ts:34]: Perhaps you could add a null check for transactions here?

Comment (komret) [reply suite-common/wallet-utils/src/transactionUtils.ts:34]: As discussed, probably better to let it fail in an unexpected case.

File: suite-common/wallet-utils/src/transactionUtils.ts [hunk 2]
```diff
@@ -267,6 +268,9 @@ export const analyzeTransactions = (
         });
     }
 
+    // make sure the known transactions are sorted properly
+    const knownSorted = known.slice(0).sort(sortByBlockHeight);
```
Comment (szymonlesisz) [inline suite-common/wallet-utils/src/transactionUtils.ts]: i totally forgot that there may be null values, so in that case it should be excluded from the list
```
known.filter(tx => tx != null).sort(sortByBlockHeight);
```

Comment (komret) [reply suite-common/wallet-utils/src/transactionUtils.ts]: feda7ed86ad0ec0c23565670970ec0d957c1186d

**Review by szymonlesisz — APPROVED** [review]
looks good to me

**Review by tomasklim — APPROVED** [review]

**Comment by bosomt** [comment]
**QA OK**

**Info:**
 - Suite version: desktop 22.9.0 (61f425c4a76a51c0e7054b1706e8383fc09772a3)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/22.9.0 Chrome/100.0.4896.160 Electron/18.3.5 Safari/537.36
 - OS: MacIntel
 - Screen: 1680x1050
 - Device: model T 2.5.2 Universal (revision 0d87b55ba4fed7eecc72bf2a94ee473830b095e9)
 - Transport: bridge 2.0.31

**Pull request merged**